extensions/ql-vscode/src/mocks/request-handlers.ts

```diff
@@ -0,0 +1,137 @@
+  const submitVariantAnalysisRequests = requests.filter(r => r.request.kind === RequestKind.SubmitVariantAnalysis);
+
+  if (submitVariantAnalysisRequests.length > 1) {
+    throw Error('More than one submit variant analysis request found');
+  }
+
+  const getRepoRequest = submitVariantAnalysisRequests[0];
+
+  return rest.post(`${baseUrl}/repositories/:controllerRepoId/code-scanning/codeql/variant-analyses`, (_req, res, ctx) => {
+    return res(
+      ctx.status(getRepoRequest.response.status),
+      ctx.json(getRepoRequest.response.body),
+    );
+  });
+}
+
+function createGetVariantAnalysisRequestHandler(requests: GitHubApiRequest[]): RequestHandler {
+  const getVariantAnalysisRequests = requests.filter(r => r.request.kind === RequestKind.GetVariantAnalysis);
+  let requestIndex = 0;
+
+  // During the lifetime of a variant analysis run, there are multiple requests
+  // to get the variant analysis. We need to return different responses for each
+  // request, so keep an index of the request and return the appropriate response.
+  return rest.get(`${baseUrl}/repositories/:controllerRepoId/code-scanning/codeql/variant-analyses/:variantAnalysisId`, (_req, res, ctx) => {
+    const request = getVariantAnalysisRequests[requestIndex];
+
+    if (requestIndex < getVariantAnalysisRequests.length - 1) {
+      // If there are more requests to come, increment the index.
+      requestIndex++;
+    }
+
+    return res(
+      ctx.status(request.response.status),
+      ctx.json(request.response.body),
+    );
+  });
+}
+
+function createGetVariantAnalysisRepoRequestHandlers(requests: GitHubApiRequest[]): RequestHandler[] {
+  const getVariantAnalysisRepoRequests = requests.filter(r => r.request.kind === RequestKind.GetVariantAnalysisRepo) as GetVariantAnalysisRepoRequest[];
+
+  const handlers: RequestHandler[] = [];
+  for (const request of getVariantAnalysisRepoRequests) {
+    const handler = rest.get(
+      `${baseUrl}/repositories/:controllerRepoId/code-scanning/codeql/variant-analyses/:variantAnalysisId/repositories/${request.request.repositoryId}`,
+      (_req, res, ctx) => {
+        return res(
+          ctx.status(request.response.status),
+          ctx.json(request.response.body),
+        );
+      });
+
+    handlers.push(handler);
+  }
+
+  return handlers;
+}
+
+function createGetVariantAnalysisRepoResultRequestHandlers(requests: GitHubApiRequest[]): RequestHandler[] {
+  const getVariantAnalysisRepoResultRequests = requests.filter(r => r.request.kind === RequestKind.GetVariantAnalysisRepoResult) as GetVariantAnalysisRepoResultRequest[];
+
+  const handlers: RequestHandler[] = [];
+  for (const request of getVariantAnalysisRepoResultRequests) {
+    const handler = rest.get(
+      `https://objects-origin.githubusercontent.com/codeql-query-console/codeql-variant-analysis-repo-tasks/:variantAnalysisId/${request.request.repositoryId}/*`,
```

Comment:
Do you think there's a way to re-use the regexes between the recorder and these request handlers? It now seems like we could easily change one of the two without changing the other.

Reply:
I've kind of been in two minds about this. I'm wondering if we can do something between gh-api-client, scenario recorded and scenario loader. I'll raise an issue to cover it.